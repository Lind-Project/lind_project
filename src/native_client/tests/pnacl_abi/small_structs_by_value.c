/*
 * Copyright (c) 2011 The Native Client Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 *
 * tests whether small structures are lowered as expected
 */

typedef struct { char x; char y; } TWO_CHARS;
typedef struct { short x; short y; } TWO_SHORTS;
typedef struct { int x; int y; } TWO_INTS;

/* @IGNORE_LINES_FOR_CODE_HYGIENE[3] */
extern void foo_chars(TWO_CHARS z);
extern void foo_shorts(TWO_SHORTS z);
extern void foo_ints(TWO_INTS z);

void bar_chars(TWO_CHARS z) {
  foo_chars(z);
}

void bar_shorts(TWO_SHORTS z) {
  foo_shorts(z);
}

void bar_ints(TWO_INTS z) {
  foo_ints(z);
}
