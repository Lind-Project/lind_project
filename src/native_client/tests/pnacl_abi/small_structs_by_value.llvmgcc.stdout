target triple = "armv7-none-linux-gnueabi"

%struct.TWO_CHARS = type { i8, i8 }
%struct.TWO_SHORTS = type { i16, i16 }
%struct.TWO_INTS = type { i32, i32 }

define void @bar_chars(%struct.TWO_CHARS* byval %z) nounwind {
entry:
  call void @foo_chars(%struct.TWO_CHARS* byval %z) nounwind
  br label %return

return:                                           ; preds = %entry
  ret void
}

declare void @foo_chars(%struct.TWO_CHARS* byval)

define void @bar_shorts(%struct.TWO_SHORTS* byval %z) nounwind {
entry:
  call void @foo_shorts(%struct.TWO_SHORTS* byval %z) nounwind
  br label %return

return:                                           ; preds = %entry
  ret void
}

declare void @foo_shorts(%struct.TWO_SHORTS* byval)

define void @bar_ints(%struct.TWO_INTS* byval %z) nounwind {
entry:
  call void @foo_ints(%struct.TWO_INTS* byval %z) nounwind
  br label %return

return:                                           ; preds = %entry
  ret void
}

declare void @foo_ints(%struct.TWO_INTS* byval)
